base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -1374,6 +1383,43 @@ boolean verifySCT(CTResponse response, byte[] tbsCert, String logPublicKey) {
         }
     }
 
+    /**
+     * parses and gleans the hashing algorithm from the returned
+     * CT signature
+     * cfu
+     */
+    /*
+       enum HashAlgorithm {
+            none, md5, sha1, sha224, sha256, sha384, sha512};
+    */
+    enum HashAlgorithm {none, MD5, SHA1, SHA224, SHA256, SHA384, SHA512};
+    public String getHashAlgFromSig(byte[] ct_signature) {
+
+        int hashingAlg = Byte.toUnsignedInt(ct_signature[0]);
+        if (hashingAlg < 4 || hashingAlg > 5)
+            return null;
+
+        return HashAlgorithm.values()[hashingAlg].name();
+    }
+
+    /**
+     * parses and gleans the signature algorithm from the returned
+     * CT signature
+     * cfu
+     */
+    /*
+       enum SignatureAlgorithm { anonymous, rsa, dsa, ecdsa}
+    */
+    enum SignatureAlgorithm {anonymous, RSA, DSA, EC};
+    public String getSigAlgFromSig(byte[] ct_signature) {
+
+        int signingAlg = Byte.toUnsignedInt(ct_signature[1]);
+        if (signingAlg < 1 || signingAlg > 3)
```

Comment:
Question: Are all 3 Signature Algorithms supported?

Reply:
both RSA and EC are.  DSA  I'm unsure, but I'm sure it'll blow up later if we don't

Reply:
The current logic seems to accept all 3 values. So, if you want disallow dsa, you might want to change the logic.

Reply:
I dont' think we should exclude that.  JSS currently has code for SHA1withDSA (it has not been updated for a long time).  If someone passes in SHA256 with DSA it would certainly fail down the road.